src/regmod/models/tobit.py

```diff
@@ -0,0 +1,238 @@
+
+        # Use JAX data structures
+        self.mat = [jnp.asarray(mat) for mat in self.mat]
+        self.uvec = jnp.asarray(self.uvec)
+        self.gvec = jnp.asarray(self.gvec)
+        self.linear_uvec = jnp.asarray(self.linear_uvec)
+        self.linear_gvec = jnp.asarray(self.linear_gvec)
+        self.linear_umat = jnp.asarray(self.linear_umat)
+        self.linear_gmat = jnp.asarray(self.linear_gmat)
+
+    @partial(jit, static_argnums=(0,))
+    def objective(self, coefs: ArrayLike) -> float:
+        """Get negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        float
+            Negative log likelihood.
+
+        """
+        weights = self.data.weights*self.data.trim_weights
+        obj_param = weights*self.get_nll_terms(coefs)
+        return obj_param.sum() + self.objective_from_gprior(coefs)
+
+    @partial(jit, static_argnums=(0,))
+    def gradient(self, coefs: ArrayLike) -> DeviceArray:
+        """Get gradient of negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        DeviceArray
+            Gradient of negative log likelihood.
+
+        """
+        return grad(self.objective)(coefs)
+
+    @partial(jit, static_argnums=(0,))
+    def hessian(self, coefs: ArrayLike) -> DeviceArray:
+        """Get hessian of negative log likelihood wrt coefficients.
+
+        Parameters
+        ----------
+        coefs : array_like
+            Model coefficients.
+
+        Returns
+        -------
+        DeviceArray
+            Hessian of negative log likelihood.
+
+        """
+        return hessian(self.objective)(coefs)
+
+    @partial(jit, static_argnums=(0,))
+    def nll(self, params: List[ArrayLike]) -> DeviceArray:
+        """Get terms of negative log likelihood wrt parameters.
+
+        Parameters
+        ----------
+        params : list[array_like]
+            Model parameters.
+
+        Returns
+        -------
+        DeviceArray
+            Terms of negative log likelihood.
+
+        """
+        vals = {
+            "mu": params[0],
+            "sigma": params[1],
+            "y": self.data.obs,
+            "nll_terms": jnp.zeros(self.data.num_obs)
+        }
+        vals = lax.fori_loop(0, self.data.num_obs, _nll_term, vals)
+        return vals["nll_terms"]
+
+    @partial(jit, static_argnums=(0,))
+    def dnll(self, params: List[ArrayLike]) -> List[DeviceArray]:
+        """Get derivative of negative log likelihood wrt parameters.
+
+        Parameters
+        ----------
+        params : list[array_like]
+            Model parameters.
+
+        Returns
+        -------
+        list[DeviceArray]
+            Derivatives of negative log likelihood.
+
+        """
+        return grad(lambda pars: jnp.sum(self.nll(pars)))(params)
```

Comment:
Can we directly redefine `jacobian2`?

Reply:
Probably, we can put this on the wishlist